internal/cli/atlas/deployments/start.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployments
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/setup"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+)
+
+type StartOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	options.DeploymentOpts
+	store  store.ClusterStarter
+	config setup.ProfileReader
+}
+
+var startTemplate = "Starting deployment '{{.Name}}'.\n"
+
+var ErrDeploymentIsDeleting = errors.New("deployment state is DELETING")
+
+func (opts *StartOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *StartOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if strings.ToLower(opts.DeploymentType) == options.LocalCluster {
+		return opts.RunLocal(ctx)
+	}
+
+	return opts.RunAtlas()
+}
+
+func (opts *StartOpts) RunLocal(ctx context.Context) error {
+	localDeployments, err := opts.GetLocalDeploymentsWithContainers(ctx)
```

Comment:
can we also check if Podman is ready here? If podman is not ready and flag == local, we should throw an error

Reply:
`GetLocalDeploymentsWithContainers` runs `opts.PodmanClient.Ready` so this is already cover

Reply:
awesome, thanks!!